ignite/distributed/comp_models/native.py

```diff
@@ -0,0 +1,204 @@
+import os
+import subprocess
+from typing import Optional, Union
+
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+
+from ignite.distributed.comp_models.base import ComputationModel
+
+
+class _NativeDistModel(ComputationModel):
+    """Private class for PyTorch native distributed computation model.
+
+    Supported `backends <https://pytorch.org/docs/stable/distributed.html#backends>`_:
+
+    - NCCL
+    - GLOO
+    - MPI
+
+    In this implementation we assume the following mapping between backend and devices:
+
+    - NCCL <-> GPU
+    - GLOO <-> CPU
+    - MPI  <-> CPU
+
+    """
+
+    name = "native-dist"
+    available_backends = tuple(
+        name
+        for name in [dist.Backend.NCCL, dist.Backend.GLOO, dist.Backend.MPI]
+        if getattr(dist, "is_{}_available".format(name))
```

Comment:
This returns a list of backends which have "is_{}_available" method even if `dist.is_{}_available() == False`. Missing `()` to call the method ?
```
if getattr(dist, "is_{}_available".format(name))()
```